pep-0691.rst

```diff
@@ -0,0 +1,452 @@
+  a repository wants to remove support for HTML.
+
+  Another option could be moving all of the existing HTML URLs under a namespace while
+  making a new namespace for JSON. Since ``/<project>/`` was defined, we would have to
+  make these namespaces not valid project names, so something like ``/_html/`` and
+  ``/_json/`` could work, then just redirect the non namespaced URLs to whatever the
+  "default" for that repository is (likely HTML, unless they've disabled HTML then JSON).
+- With separate URLs, there's no good way to support zero configuration discovery
+  that a repository supports the JSON URLs without making additional HTTP requests to
+  determine if the JSON URL exists or not.
+
+  The most naive implementation of this would be to request the JSON URL and fall back
+  to the HTML URL for *every* single request, but that would be horribly performant
+  and violate the goal of minimal additional HTTP requests.
+
+  The most likely implementation of this would be to make some sort of repository level
+  configuration file that somehow indicates what is supported. We would have the same
+  namespace problem as above, with the same solution, something like ``/_config.json``
+  or so could hold that data, and a client could first make an HTTP request to that,
+  and if it exists pull it down and parse it to learn about the capabilities of this
+  particular repository.
+- The use of ``Accept`` also allows us to add versioning into this field
+
+All being said, it is the opinion of this PEP that those three issues combined make
+using separate API routes a less desirable solution than relying on content
+negotiation to select the most ideal representation of the data.
+
+
+Appendix 1: Survey of use cases to cover
+========================================
+
+This was done through a discussion between ``pip`` and ``bandersnarch``
+maintainers, who are the two first potential users for the new API. This is
+how they use the Simple + JSON APIs today:
+
+- ``pip``:
+
+  - List of all files for a particular release
+  - Metadata of each individual artifact:
+
+    - was it yanked? (`data-yanked`)
+    - what's the python-requires? (`data-python-requires`)
+    - what's the hash of this file? (currently, hash in URL)
+    - Full metadata (`data-dist-info-metadata`)
+    - [Bonus] what are the declared dependencies, if available (list-of-strings, null if unavailable)?
+
+- ``bandersnatch`` - Only uses legacy JSON API + XMLRPC today:
+
+  - Generates Simple HTML rather than copying from PyPI
+
+    - Maybe this changes with the new API and we verbatim pull these API assets from PyPI
+
+  - List of all files for a particular release.
+
+    - Workout URL for release files to download
+
+  - Metadata of each individual artifact.
+
+    - Write out the JSON to mirror storage today (disk/S3)
+
+        - Required metadata used (via Package class - https://github.com/pypa/bandersnatch/blob/main/src/bandersnatch/package.py):
```

Comment:
This renders a bit weirdly: https://pep-previews--2578.org.readthedocs.build/pep-0691/#appendix-1-survey-of-use-cases-to-cover

Reply:
I'm not sure what's causing this. Is this a limitation of rST, an expected style for PEPs, or something syntactically wrong?

Reply:
I think it's indented by one space too many.

Reply:
I applied the change, will confirm it looks OK in the preview and then merge.